ui_framework/components/color_picker/color_picker.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { ChromePicker } from 'react-color';
+
+export class KuiColorPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showColorSelector: false,
+    };
+  }
+
+  closeColorSelector = () => {
+    this.setState({ showColorSelector: false });
+  };
+
+  toggleColorSelector = () => {
+    this.setState({ showColorSelector: !this.state.showColorSelector });
+  };
+
+  handleColorSelection = (color) => {
+    this.props.onChange(color.hex);
+  };
+
+  render() {
+    const color = this.props.color || '#ffffff';
+    const classes = classNames('kuiColorPicker', this.props.className);
+    return (
+      <div
+        className={ classes }
+        aria-label={ this.props['aria-label'] }
+        data-test-subj={ this.props['data-test-subj'] }
+      >
+        <div
+          className="kuiColorPicker__preview"
+          onClick={ this.toggleColorSelector }
+        >
+          <div
+            className="kuiColorPicker__swatch"
+            aria-label="Select a color"
```

Comment:
I also forgot to mention I love these aria-labels!! Accessibility out of the box.

Can we actually change this to be `aria-label={ this.props['aria-label'] }` and add a `defaultProp` for `ariaLabel` of "Select a color"?

The reason is that if someone provides an `aria-label` to this component in the current implementation, it will be applied to the root element. Due to the way screen readers interpret this attribute, only that root element's aria-label value will be read aloud, and the other aria-labels that belong to the child elements will be ignored entirely.